[farhan-navas] iP

farhan-navas opened the pull request:
# Eva Application

> "Your mind is for having ideas, not holding them." – David Allen

Eva is a **text-based** task management application designed to free your mind from remembering tasks. It's:

- Easy to learn
- Fast to use
- Free

## Features

- [x] Manage tasks
- [ ] Manage deadlines (coming soon)
- [ ] Reminders (coming soon)

## Getting Started

To start using Eva:

1. [Download the latest version](https://example.com/eva-download).
2. Double-click the downloaded file.
3. Add your tasks.
4. Let Eva manage your tasks for you. 😉

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

farhan-navas committed:
complete eva lvl-0

farhan-navas committed:
complete level-2

farhan-navas committed:
complete level-1

farhan-navas committed:
Merge branch 'master' of github.com:farhan-navas/ip
Accidentally labelled level-1 as level-2

farhan-navas committed:
complete level-2

farhan-navas committed:
Complete level-3

farhan-navas committed:
complete level-4

farhan-navas committed:
complete a-textuitesting

farhan-navas committed:
complete level-5

farhan-navas committed:
complete level-6

farhan-navas committed:
load tasks from hard-drive

farhan-navas committed:
save tasks to hard drive

farhan-navas committed:
integrate datetime

farhan-navas committed:
complete A-moreOOP

farhan-navas committed:
split into packages

farhan-navas committed:
Merge remote-tracking branch 'origin/add-gradle-support'
merge add-gradle-support branch

farhan-navas committed:
use gradle as build tool

farhan-navas committed:
Add JUnit tests

farhan-navas committed:
edit build.gradle to build JAR file

farhan-navas committed:
Add javadoc comments to all classes

farhan-navas committed:
Follow coding standard

farhan-navas committed:
complete find

farhan-navas committed:
Merge branch 'branch-A-JavaDoc'
Add JavaDoc changes

farhan-navas committed:
Resolve merge conflict

farhan-navas committed:
Manually resolve merge issues

farhan-navas committed:
Merge branch 'branch-Level-9'
Add find functionality

farhan-navas committed:
Add JavaDoc for find

veehz commented on src/main/java/eva/tasks/Deadline.java:
```diff
@@ -0,0 +1,74 @@
+    /**
+     * Creates a new <code>Deadline</code> object with the specified name, status and deadline.
+     *
+     * @param name name of the deadline
+     * @param isDone status of the deadline
+     * @param endTime deadline of the deadline
```
Should these have punctuation at the end of the line?
```suggestion
     * @param name name of the deadline.
     * @param isDone status of the deadline.
     * @param endTime deadline of the deadline.
```

[Link to Relevant Line in Guide](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Punctuation%20behind%20each%20parameter%20description.)

veehz commented on src/main/java/eva/Eva.java:
```diff
@@ -0,0 +1,65 @@
+ * @since 2025-09-01
+ */
+public class Eva {
+    public static ArrayList<Task> taskList  = new ArrayList<>();
```
Perhaps you can remove the extra space before "="?
```suggestion
    public static ArrayList<Task> taskList = new ArrayList<>();
```

TanJieHaoAmos replied:
> Perhaps you can remove the extra space before "="?

Same as what veehz has mentioned, could remove the extra whitespace to make the code cleaner

veehz commented on src/main/java/eva/tasks/Event.java:
```diff
@@ -0,0 +1,93 @@
+     * @param endTime end time of the event.
+     */
+    private Event(String name, LocalDate startTime, LocalDate endTime) {
+       super(name);
+       this.startTime = startTime;
+       this.endTime = endTime;
```
I think the indentation should be 8 here?
```suggestion
        super(name);
        this.startTime = startTime;
        this.endTime = endTime;
```

TanJieHaoAmos replied:
I believe the indentation should be 8 here, you may have missed an extra whitespace while indenting your code.

veehz commented on src/main/java/eva/ui/Ui.java:
```diff
@@ -0,0 +1,144 @@
+ * Represents the user interface of the program. Handles user input and prints messages to the user.
+ */
+public class Ui {
+    private final String INTRO = "Hello! I'm Eva. \nWhat can I do for you? \n";
+    private final String END = "Bye. Hope to see you again soon!";
```
Perhaps a more descriptive variable name might be better here? In case `END` may cause confusion for some.
```suggestion
    private final String INTRO_MESSAGE = "Hello! I'm Eva. \nWhat can I do for you? \n";
    private final String END_MESSAGE = "Bye. Hope to see you again soon!";
```

veehz reviewed (commented):
Looks good but there are a few nits to fix. There are some coding standard problems. Overall the naming is good too.

farhan-navas committed:
Follow coding standard

farhan-navas committed:
Follow Checkstyle

farhan-navas committed:
Create temp gui

farhan-navas committed:
Temp working version of ui

farhan-navas committed:
complete level-10

farhan-navas committed:
Add bg image

farhan-navas committed:
use varargs

TanJieHaoAmos reviewed (commented):
Overall great job! Class, method and variables name all speak for themselves so that's great, code looks clean so far 👍.

heyitssakinah commented on src/main/java/eva/ui/Ui.java:
```diff
@@ -0,0 +1,141 @@
+     * @param taskList The list of tasks to be processed.
+     * @throws TaskException If there is an error processing the tasks.
+     */
+    public String[] handleInput(String currInput, ArrayList<Task> taskList) throws TaskException {
```
Would it help improve readability to use enumerations for the valid inputs?

heyitssakinah commented on src/main/java/eva/tasks/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+     * @return new <code>Deadline</code> object.
+     * @throws TaskException if the task description is in an invalid format.
+     */
+    public static Deadline createDeadline(String taskDesc) throws TaskException {
```
nice use of factory class!

heyitssakinah reviewed (commented):
LGTM! Just some nits

farhan-navas committed:
add java assert

farhan-navas committed:
update code quality

farhan-navas committed:
replace loops with streams

farhan-navas committed:
Merge pull request #1 from farhan-navas/branch-A-Assertions

add java assert

farhan-navas committed:
Merge branch 'master' into branch-A-CodeQuality

farhan-navas committed:
Merge pull request #2 from farhan-navas/branch-A-CodeQuality

update code quality

farhan-navas committed:
Merge branch 'master' into branch-A-Streams

farhan-navas committed:
Merge pull request #3 from farhan-navas/branch-A-Streams

replace loops with streams

farhan-navas committed:
minor changes made

After completing all 3 merges required, a few minor checkstyle issues popped up which I had not noticed during the merge. This commit ensures code quality.

farhan-navas committed:
add github actions CI

farhan-navas committed:
minor changes

some checkstyle issues and test issues. led to github ci errors.

farhan-navas committed:
add C-sort extension

farhan-navas committed:
create user guide

farhan-navas committed:
improve readme styling

farhan-navas committed:
update javadoc comments

farhan-navas committed:
add additional tests

farhan-navas committed:
minor changes to event test

farhan-navas committed:
edit test file

farhan-navas committed:
fix checkstyle issues
